Server/JokesOnYou.Web.Api/Repositories/JokesRepository.cs

```diff
@@ -35,5 +35,40 @@ public async Task<IEnumerable<JokeReplyDto>> GetAllJokeDtosAsync()
 
             return jokeDtos;
         }
+
+        public async Task<Joke> GetJokeToUpdate(JokeUpdateDTO jokeUpdateDTO)
+        {
+            var jokeToUpdate = await GetJokeById(jokeUpdateDTO.Id);
+            
+            //var jokeUpdateDto = await _context.Jokes.ProjectTo<JokeUpdateDTO>(_mapper.ConfigurationProvider)
+            //    .FirstOrDefaultAsync(x => x.Id == jokeUpdateDTO.Id);
+
+            return jokeToUpdate;
+        }
+
+        public async Task Save()
+        {
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<JokeReplyDto> GetJokeReplyById(int id)
+        {
+            var joke = await GetJokeById(id); 
+
+            if (joke == null)
+            {
+                throw new NotImplementedException("can't find joke"); 
+            }
+
+            var jokeReplyDto = _mapper.Map<Joke, JokeReplyDto>(joke);
```

Comment:
_mapper.Map<Joke, JokeReplyDto> its enough to specify which entity we are mapping to which is JokeReplyDto